apps/files/helpers.py

```diff
@@ -335,30 +342,143 @@ def get_url(self, short):
+    _files = None
+
+    @property
+    def files(self):
         """
         Get the files from the primary and:
         - remap any diffable ones to the compare url as opposed to the other
         - highlight any diffs
         """
-        left_files = self.left.get_files()
-        right_files = self.right.get_files()
-        different = []
-        for key, file in left_files.items():
-            file['url'] = self.get_url(file['short'])
-            diff = file['md5'] != right_files.get(key, {}).get('md5')
-            file['diff'] = diff
-            if diff:
-                different.append(file)
-
-        # Now mark every directory above each different file as different.
-        for diff in different:
-            for depth in range(diff['depth']):
-                key = '/'.join(diff['short'].split('/')[:depth + 1])
-                if key in left_files:
-                    left_files[key]['diff'] = True
-
-        return left_files
+        if not self._files:
+            self.right.files = self._remap_filenames()
+
+            left_files = self.left.files
+            right_files = self.right.files
+            different = []
+            for key, file in left_files.items():
+                file['url'] = self.get_url(file['short'])
+                diff = file['md5'] != right_files.get(key, {}).get('md5')
+                file['diff'] = diff
+                if diff:
+                    different.append(file)
+
+            # Now mark every directory above each different file as different.
+            for diff in different:
+                for depth in range(diff['depth']):
+                    key = '/'.join(diff['short'].split('/')[:depth + 1])
+                    if key in left_files:
+                        left_files[key]['diff'] = True
+
+            self._files = left_files
+        return self._files
+
+    @memoize(prefix='file-viewer-remap-files', time=60 * 60)
+    def _remap_filenames(self):
+        """
+        Remap the filenames of add-ons whose directory contents have
+        been re-arranged. Currently handles SDK add-ons moving from
+        the `cfx` packaging format to the `jpm` format.
+        """
+        left = self.left.files
+        right = self.right.files
+
+        if u'harness-options.json' not in right or u'package.json' not in left:
+            return right
+
+        try:
+            with storage.open(right[u'harness-options.json']['full']) as f:
+                harness_options = json.load(f)
+
+            with storage.open(left['package.json']['full']) as f:
+                package = json.load(f)
+
+            assert isinstance(harness_options['mainPath'], basestring)
+            assert '/' in harness_options['mainPath']
+        except:
+            # Any errors here, just bail. Something is malformed in
+            # the add-on, and there's nothing we can do about it.
+            return right
+
+        main = '%s.js' % harness_options['mainPath'].replace('/', '/lib/')
```

Comment:
Actually, don't you simply want to prefix the path with `/lib/`? If that's the case, then it would be clearer to simply do something like ```/lib%s.js' % harness_options['mainPath']```